r/tools/nixlibs.R

```diff
@@ -435,13 +435,22 @@ turn_off_all_optional_features <- function(env_var_list) {
   # Because these are done as environment variables (as opposed to build flags),
   # setting these to "OFF" overrides any previous setting. We don't need to
   # check the existing value.
+  # Some features turn on other features (e.g. engine -> substrait -> protobuf),
+  # So the list of things to turn off is long. See:
+  # https://github.com/apache/arrow/blob/master/cpp/cmake_modules/ThirdpartyToolchain.cmake#L275
   turn_off <- c(
     "ARROW_MIMALLOC" = "OFF",
     "ARROW_JEMALLOC" = "OFF",
     "ARROW_JSON" = "OFF",
     "ARROW_PARQUET" = "OFF", # depends on thrift
     "ARROW_DATASET" = "OFF", # depends on parquet
     "ARROW_S3" = "OFF",
+    "ARROW_GCS" = "OFF",
```

Comment:
These already are OFF by default. Why do we need them OFF here too?

Reply:
If the user has set ARROW_GCS to ON and we've reached this function, the build is going to fail unless we turn it OFF. But maybe that's better than turning it off and providing a successful build that doesn't include that component? Let me know what you think behavior should be here.